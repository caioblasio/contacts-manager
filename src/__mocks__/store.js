const contactsState = require("providers/Contacts/reducer").initialState;

const history = require("history").createBrowserHistory();
const configureStore = require("store").default;

const rootState = {
  contacts: contactsState
};

const store = configureStore(rootState);

module.exports = store;
