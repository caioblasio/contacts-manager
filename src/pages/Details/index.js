import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { back } from "providers/Router/actions";
import Typography from "@material-ui/core/Typography";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ContactDetails from "containers/ContactDetails";
import ContactList from "containers/ContactList";
import useStyles from "./styles";

export const Details = ({ contactId, back }) => {
  const classes = useStyles();
  return (
    <section className={classes.root}>
      <div className={classes.back} onClick={back}>
        <ArrowBackIosIcon />
        <Typography variant="h6" component="span">
          Back
        </Typography>
      </div>
      <section className={classes.content}>
        <section className={classes.list}>
          <ContactList shortView isScrollable />
        </section>
        <section className={classes.details}>
          <Typography variant="h6" component="h1" className={classes.title}>
            Contact Details
          </Typography>
          <ContactDetails id={contactId} key={contactId} />
        </section>
      </section>
    </section>
  );
};

const mapDispatchToProps = dispatch => ({
  back: () => dispatch(back())
});

Details.propTypes = {
  back: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(Details);
