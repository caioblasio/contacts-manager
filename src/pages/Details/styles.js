import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  root: {
    ...theme.mixins.gutters(),
    [theme.breakpoints.up("sm")]: {
      paddingLeft: 0,
      paddingRight: 0
    }
  },
  content: {
    display: "flex"
  },
  list: {
    flex: "1 0 0",
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  title: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2)
  },
  details: {
    flex: "2 0 0",
    [theme.breakpoints.up("sm")]: {
      paddingRight: theme.spacing(4),
      paddingLeft: theme.spacing(4)
    }
  },
  back: {
    display: "flex",
    alignItems: "center",
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
    [theme.breakpoints.up("sm")]: {
      display: "none"
    }
  }
}));

export default useStyles;
