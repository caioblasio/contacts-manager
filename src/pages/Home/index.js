import React from "react";
import ContactList from "containers/ContactList";
import { Link } from "react-router-dom";
import Fab from "@material-ui/core/Fab";
import AddIcon from "@material-ui/icons/Add";
import Typography from "@material-ui/core/Typography";
import useStyles from "./styles";

const Home = () => {
  const classes = useStyles();

  return (
    <section className={classes.root}>
      <Typography variant="h6" component="h1" className={classes.title}>
        All Contacts
      </Typography>
      <ContactList />
      <Link to="/new" style={{ textDecoration: "none", color: "inherit" }}>
        <Fab color="primary" aria-label="add" className={classes.floatButton}>
          <AddIcon />
        </Fab>
      </Link>
    </section>
  );
};
export default Home;
