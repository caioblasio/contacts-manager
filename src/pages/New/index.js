import React from "react";
import NewContact from "containers/NewContact";
import Typography from "@material-ui/core/Typography";
import useStyles from "./styles";

const New = () => {
  const classes = useStyles();

  return (
    <section className={classes.root}>
      <Typography variant="h6" component="h1" className={classes.title}>
        New Contact
      </Typography>
      <NewContact />
    </section>
  );
};
export default New;
