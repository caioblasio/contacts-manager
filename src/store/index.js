import { createStore, applyMiddleware, compose } from "redux";
import { createBrowserHistory } from "history";
import { routerMiddleware } from "connected-react-router/immutable";
import createSagaMiddleware from "redux-saga";
import createRootReducer from "./reducers";
import rootSaga from "./sagas";

export const history = createBrowserHistory();

export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};
  if (ENVIRONMENT !== "production" && typeof window === "object") {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: "Redux",
        realtime: true,
        trace: true,
        traceLimit: 25
      });
    }
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const middlewares = [sagaMiddleware, routerMiddleware(history)];
  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createRootReducer(history),
    initialState,
    composeEnhancers(...enhancers)
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
