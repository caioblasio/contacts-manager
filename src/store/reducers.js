import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router/immutable";
import contactsReducer from "providers/Contacts/reducer";

const createRootReducer = history =>
  combineReducers({
    contacts: contactsReducer,
    router: connectRouter(history)
  });
export default createRootReducer;
