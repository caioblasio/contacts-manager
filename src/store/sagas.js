import { all } from "redux-saga/effects";
import contactsSaga from "providers/Contacts/saga";

export default function*() {
  yield all([contactsSaga()]);
}
