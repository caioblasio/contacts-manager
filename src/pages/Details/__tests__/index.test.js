import React from "react";
import { Provider } from "react-redux";
import { render, fireEvent, queryByText } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Details as DetailsComponent } from "pages/Details";

import store from "__mocks__/store";

describe("Given <DetailsComponent />", () => {
  let defaultProps;
  let component;
  let Page;

  defaultProps = {
    back: () => {}
  };

  beforeAll(() => {
    Page = props => (
      <Provider store={store}>
        <MemoryRouter>
          <DetailsComponent {...defaultProps} {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  describe("When it initializes", () => {
    beforeEach(() => {
      const { container } = render(<Page />);
      component = container;
    });

    it("Then should match the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe("When clicks on back", () => {
    let back;
    beforeEach(() => {
      back = jest.fn();
      const { container } = render(<Page back={back} />);
      component = container;
    });

    it("Then should call the back function", () => {
      fireEvent.click(queryByText(component, "Back"));
      expect(back).toHaveBeenCalled();
    });
  });
});
