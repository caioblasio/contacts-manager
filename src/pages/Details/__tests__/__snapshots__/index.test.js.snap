// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <DetailsComponent /> When it initializes Then should match the snapshot 1`] = `
<div>
  <section
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-back-6"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <span
        class="MuiTypography-root MuiTypography-h6"
      >
        Back
      </span>
    </div>
    <section
      class="makeStyles-content-2"
    >
      <section
        class="makeStyles-list-3"
      >
        <section
          class="makeStyles-root-46 makeStyles-scrollable-47"
        />
      </section>
      <section
        class="makeStyles-details-5"
      >
        <h1
          class="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
        >
          Contact Details
        </h1>
        Contact not found! :(
      </section>
    </section>
  </section>
</div>
`;
